taxonomy: kopytka

moon-rabbitOFF opened the pull request:
### What
taxonomy: kopytka + incompatibility fixes + ketchup edits

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -94787,6 +94774,7 @@ intake24_category_code:en: PTNS
 pnns_group_2:en: Cereals
 wikidata:en: Q178
 wikipedia:en: https://en.wikipedia.org/wiki/Pasta
+opposite:en: en:pasta-dishes
```
How is this the _opposite_ of pasta dishes? Did you mean to use `incompatible_with`?
```suggestion
incompatible_with:en: en:pasta-dishes
```

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -96923,13 +96911,14 @@ hr: Jela od tjestenine
 sv: Pastarätter
 intake24_category_code:en: PTDH
 wikidata:en: Q18679685
+opposite:en: en:pastas
```
```suggestion
incompatible_with:en: en:pastas
```

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -86560,6 +86546,7 @@ hr: Mesnih pripravaka
 it: Preparazioni a base di carne
 lt: Mėsos ruošiniai
 nl: Vleesbereidingen
+opposite:en: en:meals-with-meat, en:meats
```
```suggestion
incompatible_with:en: en:meals-with-meat, en:meats
```

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -84899,6 +84884,7 @@ food_groups:en: en:meat-other-than-poultry
 intake24_category_code:en: MEAT
 # Products that are poultry hopefully have the en:Poultries category which will assign the en:Poultry food group
 pnns_group_2:en: Meat
+opposite:en: en:meals-with-meat, en:meat-preparations
```
```suggestion
incompatible_with:en: en:meals-with-meat, en:meat-preparations
```

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -80396,6 +80380,7 @@ ru: Блюда с мясом
 sv: Rätter med kött
 zh: 肉餐
 intake24_category_code:en: MDSH
+opposite:en: en:meats, en:meat-preparations
```
```suggestion
incompatible_with:en: en:meats, en:meat-preparations
```

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -78960,12 +78950,7 @@ ciqual_food_code:en: 11008
-gpc_category_code:en: 10006325
-gpc_category_description:en: Definition: Includes any products that can be described/observed as a tomato Ketchup or Ketchup substitute, which is usually used to enhance the flavour, appearance and texture of savoury foods. These products can be blended with other ingredients. These products have been treated or packaged in such a way as to extend their consumable life. Definition Excludes: Speci ...
-gpc_category_name:en: Tomato Ketchup/Ketchup Substitutes (Shelf Stable)
+opposite:en: en:ketchup-and-mayonnaise-mix-sauces
```
Why are you removing `en:Tomato ketchup substitutes`?

```suggestion
incompatible_with:en: en:ketchup-and-mayonnaise-mix-sauces

< en:Ketchup
en: Tomato ketchup substitutes
gpc_category_code:en: 10006325
gpc_category_description:en: Definition: Includes any products that can be described/observed as a tomato Ketchup or Ketchup substitute, which is usually used to enhance the flavour, appearance and texture of savoury foods. These products can be blended with other ingredients. These products have been treated or packaged in such a way as to extend their consumable life. Definition Excludes: Specifically excludes Mustard/Mustard Substitutes and other Sauces Dipping/Condiments/Savoury Spreads/ Marinades, Mayonnaise/Mayonnaise Substitutes. Excludes products such as Cooking Sauces; Dressings and Dips; Frozen and Perishable Sauces Dipping, Condiments, Savoury Toppings, Savoury Spreads and Marinades.
gpc_category_name:en: Tomato Ketchup/Ketchup Substitutes (Shelf Stable)
```

moon-rabbitOFF replied:
Because it would be better to create "carrot ketchup", "beetroot ketchup"... when we have enough of those products than having a category "tomato ketchup substitutes". There is no reason to group carrot and beetroot ketchup by opposition to the tomato one in the OFF taxonomy logic, which is different than the GPC one.

Freso replied:
Yeah, @stephanegigandet answered on Slack:
> When we get products through GS1, they include a GPC category, we use those properties to map them to the OFF categories

So I don’t think they should be removed. Carrot, beetroot, etc. non-tomato ketchups can be added as children of `en:Tomato ketchup substitutes` when those categories get added.

aleene replied:
Feel free to embed these GPC categories better in the taxonomy. It was already a lot of work to have them added in the first place.

moon-rabbitOFF replied:
Okay I keep it for now.

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -78940,7 +78930,7 @@ ja: ブラバスソース
 wikidata:en: Q2102385
 
-< en:Tomato sauces
+< en:Sauces
```
Can you include a motivation in your PR for moving ketchup from tomato sauces to generic sauces?

moon-rabbitOFF replied:
Because < en:Tomato sauces should apply to "Tomato ketchup" and not "ketchup". I just added "tomato sauces" to "tomato ketchup".

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -75293,19 +75293,9 @@ pl: Sosy wegetariańskie, wegetariańskie sosy
-gpc_category_code:en: 10000054
-gpc_category_description:en: Definition: Includes any products that can be described/observed as a savoury liquid, paste, spread or granulated substance, which is usually served as an accompaniment to savoury foods. These products can be blended with other ingredients. These products must be refrigerated to extend their consumable life. Products include liquids and spreads flavoured with, but not ...
-gpc_category_name:en: Other Sauces Dipping/Condiments/Savoury Toppings/Savoury Spreads/Marinades (Perishable)
+en: Ketchup and mayonnaise mix sauces
+fr: Sauces mix ketchup et mayonnaise
+opposite:en: en:ketchup, en:mayonnaises
```
Why are you removing the two “other sauces” with GPC categories?

```suggestion
en: Other sauces

< en:Other sauces
en: Shelf-stable other sauces
gpc_category_code:en: 10000280
gpc_category_description:en: Definition: Includes any products that can be described/observed as a savoury liquid, paste, spread or granulated substance, which is usually served as an accompaniment to savoury foods. These products can be blended with other ingredients. These products have been treated or packaged in such a way as to extend their consumable life. Products include liquids, spreads and granules flavoured with, but not necessarily made from fish, meat, mushrooms, seafood or vegetables. Definition Excludes: Specifically excludes Tomato Ketchup/ Ketchup Substitutes, Mustard/Mustard Substitutes and Mayonnaise/Mayonnaise Substitutes. Excludes products such as Cooking Sauces, Frozen and Perishable Sauces Dipping, Condiments, Savoury Toppings, Savoury Spreads and Marinades.
gpc_category_name:en: Other Sauces Dipping/Condiments/Savoury Toppings/Savoury Spreads/Marinades (Shelf Stable)

< en:Other sauces
en: Refrigerated other sauces
gpc_category_code:en: 10000054
gpc_category_description:en: Definition: Includes any products that can be described/observed as a savoury liquid, paste, spread or granulated substance, which is usually served as an accompaniment to savoury foods. These products can be blended with other ingredients. These products must be refrigerated to extend their consumable life. Products include liquids and spreads flavoured with, but not necessarily made from fish, meat, mushrooms, sea food or vegetables. Definition Excludes: Specifically excludes Tomato Ketchup/ Ketchup Substitutes, Mustard/Mustard Substitutes and Mayonnaise/Mayonnaise Substitutes. Excludes products such as Cooking Sauces, Frozen and Shelf Stable Sauces Dipping, Condiments, Savoury Toppings, Savoury Spreads and Marinades.
gpc_category_name:en: Other Sauces Dipping/Condiments/Savoury Toppings/Savoury Spreads/Marinades (Perishable)

< en:Sauces
en: Ketchup and mayonnaise mix sauces
fr: Sauces mix ketchup et mayonnaise
incompatible_with:en: en:ketchup, en:mayonnaises
```

moon-rabbitOFF replied:
Because "Other sauces" might make sense in gpc context, if there main focus in on some sauces; but it doesn't in Open Food Facts. "Other sauces" doesn't mean anything out of context. Either we add a link to gpc_category_code:en: 10000280 or gpc_category_code:en: 10000054 to all the category that are not ketchup, mustard, mayonnaise or we remove it from the taxonomy.

stephanegigandet replied:
If there are GPC categories that we really do not want to keep, we could put their GPC codes on the parents (it's not supported in the code yet, but I could add it). That way when a producer imports a "Refrigerated other sauces", we map it to en:sauces

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -119559,6 +119548,12 @@ nl: Aardappelbereidingen
 intake24_category_code:en: WFFL
 
+# potato preparation similar to gnocchi, originating from Poland, and also popular in Lithuania and Belarus.
+< en:Potato preparations
+en: Kopytka
+pl: Kopytka
```
Maybe add an `xx` entry too, if this is the name it’s commonly known as (in LT and Belarus too)?

```suggestion
< en:Potato preparations
en: Kopytka
xx: Kopytka
pl: Kopytka
```

Freso reviewed (changes requested):
A lot of changes that seem unrelated to “kopytka”, incl. a lot of categories getting removed… ?

The “kopytka” change itself seems fine, but the rest I don’t know why they’re getting removed or moved around in the hierarchy etc.

moon-rabbitOFF committed:
Update categories.txt

moon-rabbitOFF committed:
Update categories.txt

Commit:
chore: Linting changes

hangy committed:
Merge remote-tracking branch 'origin/main' into kopytka

teolemon commented:
@copilot resolve the merge conflicts in this pull request

Copilot committed:
taxonomy: address PR feedback - fix opposite:en, restore GPC categories, add xx:Kopytka

Agent-Logs-Url: https://github.com/openfoodfacts/openfoodfacts-server/sessions/5e185814-3925-4de5-830e-08d5d428b6ec

Co-authored-by: teolemon <1689815+teolemon@users.noreply.github.com>

Copilot committed:
Merge remote-tracking branch 'origin/main' into kopytka

# Conflicts:
#	taxonomies/food/categories.txt

teolemon committed:
Merge branch 'main' into kopytka

Freso commented on taxonomies/food/categories.txt:
```diff
@@ -83571,8 +83577,9 @@ gpc_category_code:en: 10006325
 
-< en: Ketchup
-en: Tomato Ketchup
+< en: Tomato ketchup substitutes
```
```suggestion
< en: Ketchup
```

Tomato ketchup isn’t a substitute for tomato ketchup
